### PR TITLE
BrickGame: Allow player to pause

### PR DESCRIPTION
**Pull request by karolba**
Allow the player to pause a `BrickGame` session, by pressing either `P` or `Escape`.

This PR also adds a sentence about this functionality to the manual, and shows it in the context menu.

Screenshot 2023-03-09 at 10 02 09

Screenshot 2023-03-09 at 10 02 23

### CODE REVIEW

File: Userland/Games/BrickGame/BrickGame.cpp [hunk 1]
```diff
@@ -316,6 +317,21 @@ class Bricks final {
         return RenderRequest::RequestUpdate;
     }
 
+    void toggle_pause()
+    {
+        switch (m_state) {
+
```
Comment (MacDue) [inline Userland/Games/BrickGame/BrickGame.cpp:323]: Nit: Remove this newline 
```suggestion
```

File: Userland/Games/BrickGame/BrickGame.cpp [hunk 2]
```diff
@@ -431,18 +447,42 @@ void BrickGame::reset()
...
+{
+    m_brick_game->toggle_pause();
+    update();
+}
+
 void BrickGame::timer_event(Core::TimerEvent&)
 {
-    if (m_brick_game->state() == Bricks::GameState::GameOver) {
+    switch (m_brick_game->state()) {
+
```
Comment (MacDue) [inline Userland/Games/BrickGame/BrickGame.cpp:459]: Same here.

File: Userland/Games/BrickGame/BrickGame.cpp [hunk 3]
```diff
@@ -508,6 +548,30 @@ void BrickGame::paint_text(GUI::Painter& painter, int row, DeprecatedString cons
...
+    auto const margin = more_or_less_font_height * 2;
+
+    Gfx::IntRect pause_text_box {
+        entire_area_rect.x() + entire_area_rect.width() / 2,
+        entire_area_rect.y() + entire_area_rect.height() / 2,
+        0,
+        0
+    };
+
+    pause_text_box.inflate(paused_text_width + margin, more_or_less_font_height + margin);
```
Comment (MacDue) [inline Userland/Games/BrickGame/BrickGame.cpp:566]: Can this just use `centered_within()`?
```suggestion
   auto pause_text_box = Gfx::IntRect({}, { paused_text_width + margin, more_or_less_font_height + margin }).centered_within(entire_area_rect);
```

**Review by MacDue — APPROVED** [review]
Few little things otherwise LGTM

File: Userland/Games/BrickGame/BrickGame.cpp (hunk 1 above)
Comment (karolba) [reply Userland/Games/BrickGame/BrickGame.cpp]: Done

File: Userland/Games/BrickGame/BrickGame.cpp (hunk 2 above)
Comment (karolba) [reply Userland/Games/BrickGame/BrickGame.cpp]: Done

File: Userland/Games/BrickGame/BrickGame.cpp (hunk 3 above)
Comment (karolba) [reply Userland/Games/BrickGame/BrickGame.cpp]: >Can this just use `centered_within()`?

It can, thank you! Changed.

**Comment by bgianfo** [comment]
nice change, thanks!

**Pull request merged**